geode-dunit/src/distributedTest/java/org/apache/geode/test/greplogs/LogConsumerDistributedTest.java

```diff
@@ -0,0 +1,632 @@
+   */
+  @Test
+  public void CONTEXT_INITIALIZATION_FAILED_CLASSNOTFOUNDEXCEPTION_failsInLocator() {
+    given(LOCATOR, () -> logger.error(CONTEXT_INITIALIZATION_FAILED_CLASSNOTFOUNDEXCEPTION));
+
+    Result result = runTest(ExecuteTaskSync.class);
+
+    assertThat(getFailure(result))
+        .isInstanceOf(AssertionError.class)
+        .hasMessageContaining("error")
+        .hasMessageContaining(CONTEXT_INITIALIZATION_FAILED_CLASSNOTFOUNDEXCEPTION);
+  }
+
+  private static void given(TargetVM targetVM, SerializableRunnableIF task) {
+    for (VM vm : toArray(getAllVMs(), getController(), getLocator())) {
+      vm.invoke(() -> {
+        LogConsumerDistributedTest.targetVM = targetVM;
+        LogConsumerDistributedTest.task = task;
+      });
+    }
+  }
+
+  private static Throwable getFailure(Result result) {
+    assertThat(result.getFailures()).hasSize(1);
+    return result.getFailures().get(0).getException();
+  }
+
+  private static VM getTargetVm() {
+    switch (targetVM) {
+      case CONTROLLER:
+        return getController();
+      case ANY_VM:
+        return getVM(0);
+      case LOCATOR:
+        return getLocator();
+      default:
+        throw new IllegalStateException("VM for " + targetVM + " not found");
+    }
+  }
+
+  enum TargetVM {
+    CONTROLLER,
+    LOCATOR,
+    ANY_VM
+  }
+
+  public static class ExecuteTaskSync implements Serializable {
+
+    @Rule
+    public DistributedRule distributedRule = new DistributedRule();
+
+    @Test
+    public void invokeTaskInTargetVm() {
+      getTargetVm().invoke(() -> task.run());
+    }
+  }
+
+  public static class ExecuteTaskAsync implements Serializable {
+
+    private transient AsyncInvocation<Void> asyncInvocation;
+
+    @Rule
+    public DistributedRule distributedRule = new DistributedRule();
+
+    @After
+    public void tearDown() throws Exception {
+      asyncInvocation.await();
+    }
+
+    @Test
+    public void invokeTaskInTargetVm() {
+      asyncInvocation = getTargetVm().invokeAsync(() -> task.run());
+    }
+  }
+
+  public static class ExecuteTaskBefore implements Serializable {
+
+    @Rule
+    public DistributedRule distributedRule = new DistributedRule();
+
+    @Before
+    public void invokeTaskInBefore() {
+      getTargetVm().invokeAsync(() -> task.run());
+    }
+
+    @Test
+    public void doNothing() {
+      // nothing
+    }
+  }
+
+  public static class ExecuteTaskBeforeClass implements Serializable {
+
+    @BeforeClass
+    public static void invokeTaskInBeforeClass() {
+      getTargetVm().invokeAsync(() -> task.run());
+    }
+
+    @Rule
+    public DistributedRule distributedRule = new DistributedRule();
+
+    @Test
+    public void doNothing() {
+      // nothing
+    }
+  }
+
+  private static final String CONTEXT_INITIALIZATION_FAILED_CLASSNOTFOUNDEXCEPTION =
+      "[error 2019/11/04 13:09:31.730 PST <RMI TCP Connection(1)-127.0.0.1> tid=0x13] Context initialization failed"
```

Comment:
Maybe pick a less realistic looking error message - like "Test error message in the logs" or something like that?

Reply:
Fixed this. Also renamed test methods to have consistent names.